src/ocm/components/hosts/HostsDiscoveryTable.tsx

```diff
@@ -140,7 +142,8 @@ const HostsDiscoveryTable: React.FC<HostsDiscoveryTableProps> = ({
           alerts
             .filter((alert) => alert.title === 'Failed to set role')
             .forEach((a) => removeAlert(a.key));
-        } catch (error) {
+        } catch (e) {
+          const error = e as Parameters<typeof handleApiError>[0];
```

Comment:
In static IP branch I fixed the handleApiError to get unknown instead of AxiosError.
It's correct for typesscript to complain about this, when the error is a string for example, not directly an API error, this error handling fails. So after the static ip merge we won't need this casting.

Reply:
@batzionb please post a snippet of how you're fixing it in your PR and I'll commit the same changes to mine

Reply:
Look at the file  
src/ocm/api/utils.ts
and src/ocm/api/types.ts
in the [static ip PR](https://github.com/openshift-assisted/assisted-ui-lib/pull/1328/files)

![image](https://user-images.githubusercontent.com/22211154/168031195-72fd3268-8cf0-450a-bf26-fb465748e4ba.png)

With this change, when the functions handleApiError and getErrorMessage are used there are no TS errors